src/main/java/org/testng/TestNG.java

```diff
@@ -142,14 +142,16 @@
   private final Map<Class<? extends IReporter>, IReporter> m_reporters = Maps.newHashMap();
   private final Map<Class<? extends IDataProviderListener>, IDataProviderListener> m_dataProviderListeners = Maps.newHashMap();
 
+  protected static final int HAS_STARTED = -1;
+  protected static final int SUCCESS = 0;
   protected static final int HAS_FAILURE = 1;
```

Comment:
Could you try to replace these values by an enum? I think it will highlight some issues in the logic.